src/core/trie.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {type AddressModule} from "./address";
+import {
+  NodeAddress,
+  type NodeAddressT,
+  EdgeAddress,
+  type EdgeAddressT,
+} from "./graph";
+import * as NullUtil from "../util/null";
+
+type Entry<T> = {map: RecursiveMap<T>, value: ?T};
+type RecursiveMap<T> = Map<string, Entry<T>>;
+class BaseTrie<K, T> {
```

Comment:
minor: Typically, the type parameters to a key-value data structure
are `<K, V>` instead of `<K, T>`.

Reply:
Fixed